app/lib/widgets/action_selection.dart

```diff
@@ -62,18 +62,91 @@ class _ActionSelectionState extends State<ActionSelection> {
         actions.where((action) => !possibleActions.contains(action)).toList();
   }
 
+  void _showFilterDialog(BuildContext context) {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return StatefulBuilder(builder: (context, setStateDialog) {
+          return AlertDialog(
+            title: Text(AppLocalizations.of(context)!.patientScreenFilterResourceTitle),
+            content: SizedBox(
+                width: MediaQuery.of(context).size.width * 0.8,
+                child: ResourceDirectory(
+                  initiallyExpanded: true,
+                  locations: widget.locations,
+                  resourceToggle: (resource) {
+                    toggleResource(resource);
+                    setStateDialog(() {});
+                  },
+                )),
+            actions: [
+              TextButton(
+                onPressed: () {
+                  Navigator.of(context).pop();
+                },
+                child: Text(AppLocalizations.of(context)!.dialogueClose),
+              ),
+            ],
+          );
+        });
+      },
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
+
+    resources = Location.flattenResourcesFromList(widget.locations);
+
     return Column(children: [
       ApiFutureBuilder<List<PatientAction>>(
         future: futureActions,
         builder: (context, actions) {
           setActions(actions);
           var selectedActions = getSelectedActions();
           return Column(children: [
-            Text(selectedActions.isNotEmpty
-                ? AppLocalizations.of(context)!.patientActions
-                : AppLocalizations.of(context)!.patientNoActions),
+            Stack(
+              alignment: Alignment.center,
+              children: [
+                Align(
+                  alignment: Alignment.center,
+                  child: Text(
+                    AppLocalizations.of(context)!.patientActions,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+                Align(
+                  alignment: Alignment.centerRight,
+                  child: IconButton(
+                    onPressed: () => _showFilterDialog(context),
+                    icon: const Icon(Icons.filter_list),
+                  ),
+                ),
+              ],
+            ),
+            if (getSelectedResources().isNotEmpty) ...[
```

Comment:
I do really like how this change in filtering through resources turned out! All in all these changes are very good on the eyes :P